app/src/main/java/pm/gnosis/heimdall/ui/safe/recover/recoveryphrase/ConfirmNewRecoveryPhraseActivity.kt

```diff
@@ -0,0 +1,68 @@
+package pm.gnosis.heimdall.ui.safe.recover.recoveryphrase
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.rxkotlin.plusAssign
+import io.reactivex.rxkotlin.subscribeBy
+import pm.gnosis.heimdall.data.repositories.TransactionData
+import pm.gnosis.heimdall.data.repositories.models.SafeTransaction
+import pm.gnosis.heimdall.di.components.ViewComponent
+import pm.gnosis.heimdall.ui.recoveryphrase.ConfirmRecoveryPhraseActivity
+import pm.gnosis.heimdall.ui.transactions.view.review.ReviewTransactionActivity
+import pm.gnosis.model.Solidity
+import pm.gnosis.utils.asEthereumAddress
+import pm.gnosis.utils.asEthereumAddressString
+import pm.gnosis.utils.nullOnThrow
+import timber.log.Timber
+
+class ConfirmNewRecoveryPhraseActivity : ConfirmRecoveryPhraseActivity<ConfirmNewRecoveryPhraseContract>() {
+    override fun inject(component: ViewComponent) = component.inject(this)
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val safeAddress = nullOnThrow { intent.getStringExtra(EXTRA_SAFE_ADDRESS).asEthereumAddress()!! } ?: run { finish(); return }
+        val browserExtension = nullOnThrow { intent.getStringExtra(EXTRA_BROWSER_ADDRESS).asEthereumAddress()!! } ?: run { finish(); return }
+        viewModel.setup(safeAddress, browserExtension)
+    }
+
+    override fun isRecoveryPhraseConfirmed() {
+        disposables += viewModel.loadTransaction()
+            .observeOn(AndroidSchedulers.mainThread())
+            .doOnSubscribe { onLoadingTransaction(isLoading = true) }
+            .doAfterTerminate { onLoadingTransaction(isLoading = false) }
+            .subscribeBy(onSuccess = { onTransaction(it.first, it.second) }, onError = ::onTransactionError)
+    }
+
+    private fun onTransaction(safeAddress: Solidity.Address, safeTransaction: SafeTransaction) {
+        startActivity(
+            ReviewTransactionActivity.createIntent(
+                this,
+                safeAddress,
+                txData = TransactionData.ReplaceRecoveryPhrase(safeAddress, safeTransaction)
+            )
+        )
+    }
+
+    private fun onTransactionError(throwable: Throwable) {
+        Timber.e(throwable)
```

Comment:
no error message, no retry?

Reply:
Retry is done on the parent class

https://github.com/gnosis/safe-android/blob/c4ec4961f4e0226776b5948bd2e4eac8bb61884c/app/src/main/java/pm/gnosis/heimdall/ui/recoveryphrase/ConfirmRecoveryPhraseActivity.kt#L88-L91

Missing the error message yeah.